zio-http/src/main/scala/zio/http/endpoint/internal/EndpointClient.scala

```diff
@@ -32,9 +32,17 @@ private[endpoint] final case class EndpointClient[I, E, O, M <: EndpointMiddlewa
     val request0 = endpoint.input.encodeRequest(invocation.input)
     val request  = request0.copy(url = endpointRoot ++ request0.url)
 
-    val requestPatch = invocation.middleware.input.encodeRequestPatch(mi)
+    val requestPatch            = invocation.middleware.input.encodeRequestPatch(mi)
+    val patchedRequest          = request.patch(requestPatch)
+    val withDefaultAcceptHeader =
+      if (patchedRequest.headers.exists(_.headerName == Header.Accept.name))
+        patchedRequest
+      else
+        patchedRequest.addHeader(
+          Header.Accept(MediaType.parseCustomMediaType("application/protobuf").get, MediaType.application.json),
```

Comment:
@jdegoes If I understood your right in one of our 1on1's, you want that protopuf is the default accept header for the endpoint client, right? I am asking, because this fails the test currently. Since the error message that is tested for is different. I am not sure if this is a bug. But I wanted to make sure, that I understood you right.